vendor/engines/split_accounts/app/services/split_accounts/order_detail_splitter.rb

```diff
@@ -78,9 +80,9 @@ def apply_order_detail_remainders
     end
 
     def apply_reservation_remainders
-      return unless order_detail.reservation
-      applier = RemainderApplier.new(order_detail.reservation, split_order_details.map(&:reservation), splits)
-      applier.apply_remainders(reservation_attribute_splitter)
+      return unless order_detail.time_data.present?
```

Comment:
I still would learn towards `time_based_purchase` or something similar to make the relationship a little more obvious, but maybe that isn't intuitive to others.